pyomo/core/tests/unit/test_numeric_expr.py

```diff
@@ -3677,6 +3677,45 @@ def test_summation7(self):
         e = quicksum(self.m.p[i]*self.m.q[i] for i in self.m.I)
         self.assertEqual( e(), 15 )
         self.assertIs(type(e), SumExpression)
+        
+    def test_quicksum_reject_noniterable(self):
+        msg = 'Failed to recognize non-iterable args.'
+        try:
+            e = quicksum(1)
+            self.fail(msg)
+        except Exception as ex:
+            self.assertIsInstance(ex, TypeError, msg)
```

Comment:
We should use the standard assertion here:
```python
with self.assertRaisesRegex(TypeError, "'int' object is not iterable"):
    quicksum(1)
```

If you also adopt the suggestion below of logging an error message, we could test that as well:
```python
with LoggingIntercept() as LOG:
    with self.assertRaisesRegex(TypeError, "'int' object is not iterable"):
        quicksum(1)
self.assertEqual(LOG.getvalue(), 'The argument `args` to quicksum() is not iterable!\n')
```

Reply:
Thank you. All changes have been done.